build/common.jl

```diff
@@ -25,17 +35,41 @@ function remote_url(repo_root::AbstractString, name::AbstractString="origin")
     end
 end
 
-const REPO_PATH = abspath(joinpath(@__DIR__, "..", ".."))
-const PKG_URL = remote_url(REPO_PATH)
+function set_url_scheme(url, scheme)
+    m = match(LibGit2.URL_REGEX, url)
+    if m === nothing
+        throw(ArgumentError("URL is not a valid SCP or HTTP(S) URL: $(url)"))
+    end
+    return LibGit2.git_url(; scheme="https", username=something(m[:user], ""),
+                           host=something(m[:host], ""), port=something(m[:port], ""),
+                           path=something(m[:path], ""))
+end
 
-# Read Project.toml
-const JLL_PATH = joinpath(REPO_PATH, "ray_julia_jll")
-const JLL_PROJECT_TOML = joinpath(JLL_PATH, "Project.toml")
-const JLL_ARTIFACTS_TOML = joinpath(JLL_PATH, "Artifacts.toml")
+# Used to convert `HostPlatform` into something contained in
+# `BinaryBuilder.support_platforms()`
+function supported_platform(p::Platform)
+    support_tags = filter((k, v)::Pair -> k in ("call_abi", "libc"), tags(p))
+    support_tags["julia_version"] = string(Base.thisminor(VERSION))
+    support_tags = [Symbol(k) => v for (k, v) in support_tags]
+    return Platform(arch(p), platform_name(p); support_tags...)
+end
 
-const JLL_PROJECT = read_project(JLL_PROJECT_TOML)
-const TAG = "v$(JLL_PROJECT.version)"
+function gen_artifact_url(; repo_url, tag, filename)
+    return join([repo_url, "releases", "download", tag, filename], '/')
+end
 
-const GITHUB_URL = "https://api.github.com/repos"
-const _PKG = split(PKG_URL, ":")[2]
-const ASSETS_URL = joinpath(GITHUB_URL, "$_PKG", "releases", "tags", "$TAG")
+function gen_artifact_filename(; tag::AbstractString, platform::Platform)
+    return "ray_julia.$tag.$(triplet(platform)).tar.gz"
```

Comment:
```suggestion
    return "ray_julia.$tag.$(triplet(supported_platform(platform))).tar.gz"
```

I wonder do we want to call `supported_platform` again here for extra protection...although this function is only called over the `SUPPORTED_PLATFORMS` anyway it's probably overkill